### ISSUE TITLE
Bug report: block edit plus tab

### ISSUE THREAD

**Issue by LaurV2** [notepad-plus-plus/notepad-plus-plus#11834]
### Description of the Issue
When selecting a block (with shift+alt) and pressing tab, the "columns" indicator in the status bar shows random numbers instead of real column. Don't know if matters, the tab is set to 3, and "replace with spaces" (edit: I tested it, it doesn't matter!). This is mostly cosmetic, but quite bothering, as the feature is used quite a lot, for example, when you try to align comments on the rights at the end of the line, or matrices of data, etc. The bug is introduced recently, I don't know with which version, but it was not like that in some very older versions. Sorry I don't keep track and I don't upgrade often ("if it works, don't fix it"). If it matters, the bug appears in win 7 and win 10 too. 

### Steps to Reproduce the Issue

1. go anywhere in the text, hold "shift+alt", press "down arrow" a couple of times
2. release shift+alt
3. press tab a couple of times 
4. watch the columns indicator on the status bar

### Expected Behavior

I expected Notepad++ to know how to count.

### Actual Behavior

Notepad ++ didn't know how to count. Back to primary school! >:P

### Debug Information

Notepad++ v8.4.2   (64-bit)
Build time : May 29 2022 - 16:47:30
Path : C:\Notepad++\notepad++.exe
Command Line : $COMMAND_LINE_PLACEHOLDER$
Admin mode : OFF
Local Conf mode : ON
Cloud Config : OFF
OS Name : Windows 7 Professional (64-bit) 
OS Build : 7601.24443
Current ANSI codepage : 1252
Plugins : 
    mimeTools (2.8)
    NppConverter (4.4)
    NppExport (0.4)

**Comment by alankilborn** [comment]
Here's an interesting scenario.

Have some text data:

```
x

xx
xxx
xxxxx
xxxxxxxxxY
```

Put the caret after the `Y`.  Note `Col: 11`.

Press Shift+Alt+Up once to obtain:

![image](https://user-images.githubusercontent.com/77065706/175524954-6f2e6c64-bd33-4371-903c-7648c321db61.png)

Note: `Col: 6`.  This is the column of the last data on line 5, the column of the primary caret.  

Side note: When you have rectangular selection data over "n" lines, you have "n" carets, one per line; only one of these carets is the primary or main caret, and it will either be the top line selected (in this example) or the bottom, depending upon if you are doing Up or Down movements.  In my screenshots note that the primary caret is colored purple (hard to see unless the block is zero width).

Press Shift+Alt+Up again to obtain:

![image](https://user-images.githubusercontent.com/77065706/175525134-661c7479-e9d3-43d4-b4cc-c4621e6bb72f.png)

Note: `Col: 4`.  This is the column of the last data on line 4, the column of the primary caret.

Using Shift+Alt+Up repeatedly, take the blocking up to line 1.  Note `Col: 2`.  Thus far this makes reasonable sense.

Now press Shift+Alt+Right a few times.  Note that the column doesn't change, because the primary caret is now in something called _virtual space_ (to the right of the real data on a line).

Keep pressing Shift+Alt+Right until you obtain:

![image](https://user-images.githubusercontent.com/77065706/175525804-aa99bda6-5a91-4a7e-b95a-9bc9ba81e315.png)

Now press Tab; with 4-space tabbing, I obtained:

![image](https://user-images.githubusercontent.com/77065706/175526000-66789d3a-0746-4d86-8b0e-2698457e4cb8.png)

which seems reasonable.  I also obtained `Col: 5` which also seems reasonable when considering the primary caret (on line 1).

So, because the OP didn't present any data along with STR, I don't know if my scenario is representative, but it certainly isn't exhibiting any random behavior.  It's all explainable... now whether or not it is _desirable_ is another question. :-)
